app/src/main/java/io/github/pietrocaselani/moviestraker/di/AppInjector.kt

```diff
@@ -0,0 +1,59 @@
+package io.github.pietrocaselani.moviestraker.di
+
+import android.app.Activity
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentActivity
+import android.support.v4.app.FragmentManager
+import android.support.v4.app.FragmentManager.FragmentLifecycleCallbacks
+import dagger.android.AndroidInjection
+import dagger.android.support.AndroidSupportInjection
+import dagger.android.support.HasSupportFragmentInjector
+import io.github.pietrocaselani.moviestraker.ActivityLifecycleCallbacksAdapter
+import io.github.pietrocaselani.moviestraker.MoviesTrackerApplication
+import io.github.pietrocaselani.moviestraker.ui.moviedetails.MovieDetailsModule
+import io.github.pietrocaselani.moviestraker.ui.searchmovies.SearchMoviesModule
+import io.github.pietrocaselani.moviestraker.ui.upcoming.UpcomingModule
+
+/**
+ * Created by pc on 24/06/17.
+ */
+object AppInjector {
```

Comment:
First of all, I am not a Dagger master, I know the basics.
It's been a while since I have seen this approach, and I liked because it's a centralized place to create modules and to worry about inject dependencies on Android platform components.
There is some things I really don't like on this approach, for example, that empty `Injectable` interface and the need to check at runtime if the fragment supports injection (`if (fragment is Injectable)`), but I still have not stopped to think of a better solution.